[DOCS] document `Content-Type` breaking change from 8.0 (#116845)
elastic/elasticsearch: docs/reference/migration/migrate_8_0/rest-api-changes.asciidoc

*Impact* +
To detect a server timeout, check the `timed_out` field of the JSON response.
====

.The `Content-Type` response header no longer specifies the charset.
[%collapsible]
====
*Details* +

> The `Content-Type` response header no longer specifies the charset. This information is not required when transferring JSON data, as UTF is the default encoding.

Reviewer: suggested change: as JSON text shall always be encoded in Unicode, with UTF-8 being the default encoding.

> The `Content-Type` response header no longer specifies the charset. This information is not required when transferring JSON data, as UTF is the default encoding.

Author: ```suggestion
The `Content-Type` response header no longer specifies the charset. This information is not required when transferring JSON data, because JSON text is always encoded in Unicode, with UTF-8 being the default encoding.
```

> The `Content-Type` response header no longer specifies the charset. This information is not required when transferring JSON data, because JSON text is always encoded in Unicode, with UTF-8 being the default encoding.

Author: ```suggestion
The `Content-Type` response header no longer specifies the charset. This information is not required when transferring JSON data, because JSON text will always be encoded in Unicode, with UTF-8 being the default encoding.
```

*Impact* +

> Some applications and utilities, such as PowerShell's https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/invoke-restmethod[Invoke-RestMethod], must receive charset information to display data correctly. If your application or utility relies on charset information in the `Content-Type` response header, UTF-8 encoded characters will be rendered incorrectly in the response body.

Reviewer: Optional: specify this is a bug of the utilities like PowerShell

*Impact* +
To detect a server timeout, check the `timed_out` field of the JSON response.
====

.The `Content-Type` response header no longer specifies the charset.
[%collapsible]
====
*Details* +
The `Content-Type` response header no longer specifies the charset. This information is not required when transferring JSON data, as UTF is the default encoding.

*Impact* +
Some applications and utilities, such as PowerShell's https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/invoke-restmethod[Invoke-RestMethod], must receive charset information to display data correctly. If your application or utility relies on charset information in the `Content-Type` response header, UTF-8 encoded characters will be rendered incorrectly in the response body.

As a workaround, to render non-ASCII characters, include an HTTP `Accept` header in your requests, specifying the charset: 

[source,sh]
----
Accept: application/json; charset=utf-8
----

> // NOTCONSOLE

Author: ```suggestion
```
